pallets/parachain-staking/src/tests.rs

```diff
@@ -860,6 +861,132 @@ fn sufficient_join_candidates_weight_hint_succeeds() {
 		});
 }
 
+#[test]
+fn collator_goes_offline_if_doesnt_produce_blocks() {
+	ExtBuilder::default()
+		.with_balances(vec![(0, 10)])
+		.with_candidates(vec![(0, 10)])
+		.build()
+		.execute_with(|| {
+			//roll to round 3
+			roll_to_round_begin(3);
+
+			//finalize the first block of round 3
+			ParachainStaking::on_finalize(10);
+
+			//assert correct events
+			assert_events_eq!(
+				Event::CollatorChosen {
+					round: 3,
+					collator_account: 0,
+					total_exposed_amount: 10,
+				},
+				Event::NewRound {
+					starting_block: 10,
+					round: 3,
+					selected_collators_number: 1,
+					total_balance: 10,
+				},
+			);
+
+			//check that storage has updated for collator 0 when it finalized the block
+			assert_eq!(
+				ParachainStaking::candidate_rounds_info(0),
+				Some(CandidateLastRound(3))
+			);
+
+			//roll 4 rounds more
+			roll_to_round_begin(7);
+
+			//check that the collator goes offline if it doesn't produce blocks
+			//within NumberRoundsOffline rounds (in this case 3)
+			assert_events_eq!(
+				Event::CandidateWentOffline { candidate: 0 },
+				Event::CollatorChosen {
+					round: 7,
+					collator_account: 0,
+					total_exposed_amount: 10,
+				},
+				Event::NewRound {
+					starting_block: 30,
+					round: 7,
+					selected_collators_number: 1,
+					total_balance: 10,
+				},
+			);
+
+			//check the storage for the collator 0 has been removed
+			assert_eq!(ParachainStaking::candidate_rounds_info(0), None);
+		});
+}
+
+#[test]
+fn collator_produces_blocks_successfully() {
+	ExtBuilder::default()
+		.with_balances(vec![(0, 10)])
+		.with_candidates(vec![(0, 10)])
+		.build()
+		.execute_with(|| {
+			//roll to round 3
```

Comment:
Since we're testing across rounds, it makes sense to set blocks per round to something smaller. Some of the similar tests have this as the first tx, so they don't run long:

```rust
               assert_ok!(ParachainStaking::set_blocks_per_round(
			RuntimeOrigin::root(),
			6u32
		));
```